lib/components/ScheduleSelector/components/DayPicker/DayPicker.tsx

```diff
@@ -13,29 +13,26 @@ interface DayPickerProps {
 }
 
 const DayPicker = ({ days, onChange, options, isDisabled }: DayPickerProps) => {
-  const [selectedDays, setSelectedDays] = useState<string[]>(
-    days.split(',').map((day) => day.trim())
-  )
-  const toggleDaySelection = useCallback(
-    (day: string): void => {
-      if (selectedDays.includes(SELECT_ALL)) {
-        setSelectedDays([day])
-        return
-      }
-      setSelectedDays((prev) =>
-        prev.includes(day) ? prev.filter((d) => d !== day) : [...prev, day]
-      )
-    },
-    [selectedDays]
+  const allDays = useMemo(
+    () => (options || DAYS_OF_WEEK).map((day) => day.value),
+    [options]
   )
 
-  useEffect(() => {
-    setSelectedDays(days.split(',').map((day) => day.trim()))
-  }, [days])
+  const selectedDays =
+    days === 'all' ? allDays : days.split(',').map((day) => day.trim())
```

Comment:
I guess you already have 'all' as a const

Reply:
yes, I should have :)